tests/mount.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * Copyright (c) 2023 PHYTEC Messtechnik GmbH
+ */
+
+#include <glib.h>
+#include <glib/gstdio.h>
+#include <gio/gio.h>
+#include "helper.h"
+#include "pu-error.h"
+#include "pu-mount.h"
+
+#define MOUNT_POINT "/run/partup/partition"
+#define MOUNT_SOURCE "data/root.ext4"
+
+static void
+test_create_mount_point(void)
+{
+    g_autoptr(GError) error = NULL;
+    const gchar *name = "partition";
+
+    g_autofree gchar *mount_point = pu_create_mount_point(name, &error);
+    g_assert_nonnull(mount_point);
+    g_assert_cmpstr(mount_point, ==, MOUNT_POINT);
+    g_assert_no_error(error);
+    g_assert_true(g_file_test(mount_point, G_FILE_TEST_IS_DIR));
+    g_assert_cmpint(g_rmdir(mount_point), ==, 0);
+}
+
+static void
+test_mount(void)
+{
+    g_autoptr(GError) error = NULL;
+    g_autofree gchar *mount_point = NULL;
+    gint wait_status;
+
+    mount_point = g_dir_make_tmp("partup-XXXXXX", &error);
+    g_assert_no_error(error);
+
+    g_assert_true(pu_mount(MOUNT_SOURCE, mount_point, &error));
+    g_assert_no_error(error);
+
+    const gchar *cmd = g_strdup_printf("umount %s", mount_point);
+    g_assert_true(g_spawn_command_line_sync(cmd, NULL, NULL,
+                                            &wait_status, &error));
+    g_assert_true(g_spawn_check_wait_status(wait_status, &error));
+    g_assert_no_error(error);
+    g_assert_cmpint(g_rmdir(mount_point), ==, 0);
+}
+
+static void
+test_umount(void)
+{
+    g_autoptr(GError) error = NULL;
+    g_autofree gchar *mount_point = NULL;
+    mount_point = g_dir_make_tmp("partup-XXXXXX", &error);
+    g_assert_no_error(error);
+
+    const gchar *cmd = g_strdup_printf("mount %s %s", MOUNT_SOURCE, mount_point);
+    gint wait_status;
```

Comment:
Please separate declaration and initialization of variables. See also other instances.